features/out_of_memory.feature

```diff
@@ -0,0 +1,65 @@
+Feature: Out of memory errors
+
+  Background:
+    Given I clear all persistent data
+
+  Scenario: Out of memory errors are enabled when loading configuration
+    When I run "OOMLoadScenario"
+    And I wait to receive a request
+    Then the "Bugsnag-API-Key" header equals "0192837465afbecd0192837465afbecd"
+    And the event "unhandled" is false
+    And the exception "message" equals "OOMLoadScenario"
+    And the event has a "manual" breadcrumb named "OOMLoadScenarioBreadcrumb"
+    And I discard the oldest request
+
+    When I relaunch the app
+    And I configure Bugsnag for "OOMLoadScenario"
+    And I wait to receive a request
+    Then the "Bugsnag-API-Key" header equals "0192837465afbecd0192837465afbecd"
+    And the error is an OOM event
```

Comment:
At the moment I'm missing the bit of the jigsaw that makes this happen - please explain IRL :-)

Reply:
If you mean the step, it's present here: https://github.com/bugsnag/bugsnag-cocoa/pull/842/files#diff-3be9e7e3abbbdb0df337cf5d28bc9f8548167b7f6232f6b9c4b18da06e7c3012R242

Reply:
Oh, sorry - ambiguous comment.  I meant how does the OOM actually occur?  There's a sentinel file that's written during the test, right?  I can't see the code that does that, but presumably that's because I can't find OOMBaseScenario.

Reply:
That's been removed in favour of force restarting the app in the middle of a test run.  For example, [https://github.com/bugsnag/bugsnag-cocoa/pull/842/files#diff-776a4177f78bb5fb9417fbf06945d091b0956ac9a1e0cd6d94bc17601eb074aeR15](this step) will cause appium to kill and restart the app, testing the OOM triggering mechanism as well as the configured response.

While I feel this is a better solution in general as it tests the whole of the OOM process, it does mean these tests may be less compatible with non-browserstack devices, as the way appium kills the app and whether it triggers the OOM mechanism, may differ between platforms.

Reply:
Thanks, that makes a lot more sense now.  In that case, can I suggest we write a comment something like the following at the top of the feature file, and annotate each `I relaunch the app` step with "See note above"?

> Due to the combination of BrowserStack's behaviour when resetting the app and the way that our OOM detection works, the `I relaunch the app` steps are currently sufficient to trigger the OOM mechanism.  However, these tests may not behave in the same way on local devices, device farms other that BrowserStack, or if we change that OOM detection works.

Reply:
As a solution to this I'd suggest skipping the OOM test by default, but run them if a flag is present.  This will allow us to verify OOMs on CI without breaking our E2E tests running locally on device.

Reply:
I'm not a fan of skipping tests by default, in case we forget to add the necessary flag when the tests should be run.  For now there are other issues running tests on local devices (I think any unhandled errors), so this will probably come out in the wash - but in principle yes we could certainly do something to opt-out of running them on local devices.